pkg/repository/repository.go

```diff
@@ -31,13 +32,13 @@ type PackageResources struct {
 }
 
 type PackageRevisionKey struct {
-	Repository, Package, Revision string
-	WorkspaceName                 v1alpha1.WorkspaceName
+	Repository, Directory, Package, Revision string
```

Comment:
I'm not sure adding a "Directory" to a key for packages in all repo types is a good idea. "Directory" is a git repo thing.

Reply:
I think package names with a slash in them are illegal, package names must comply with kubernetes naming rules, which I think is alphanumeric and the characters "." and "-".
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/